addons/base_automation/models/base_automation.py

```diff
@@ -322,7 +322,7 @@ def write(self, vals, **kw):
                 actions = self.env['base.automation']._get_actions(self, ['on_write', 'on_create_or_write'])
                 if not (actions and self):
                     return write.origin(self, vals, **kw)
-                records = self.with_env(actions.env)
+                records = self.with_env(actions.env).filtered(lambda r: bool(r.id))
```

Comment:
You can use a field name as argument to `filtered`:
```suggestion
                records = self.with_env(actions.env).filtered('id')
```